Source/WebKit/UIProcess/API/Cocoa/_WKWebExtensionContext.h

```diff
@@ -453,6 +453,118 @@ WK_CLASS_AVAILABLE(macos(13.3), ios(16.4))
+ @seealso didCloseWindow:
+ @seealso openWindows
+ */
+- (void)didOpenWindow:(id <_WKWebExtensionWindow>)newWindow;
+
+/*!
+ @abstract Should be called by the app when a window is closed to fire appropriate events with only this extension.
+ @param newWindow The window that was closed.
+ @discussion This method informs only the specific extension of the closure of a window. If the intention is to inform all loaded 
+ extensions consistently, you should use the respective method on the extension controller instead.
+ @seealso didOpenWindow:
+ @seealso openWindows
+ */
+- (void)didCloseWindow:(id <_WKWebExtensionWindow>)closedWindow;
+
+/*!
+ @abstract Should be called by the app when a window gains focus to fire appropriate events with only this extension.
+ @param focusedWindow The window that gained focus, or \c nil if no window has focus or a window has focus that is not visible to this extension.
+ @discussion This method informs only the specific extension that a window has gained focus. If the intention is to inform all loaded
+ extensions consistently, you should use the respective method on the extension controller instead.
+ */
+- (void)didFocusWindow:(nullable id <_WKWebExtensionWindow>)focusedWindow;
+
+/*!
+ @abstract Should be called by the app when a new tab is opened to fire appropriate events with only this extension.
+ @param newTab The newly opened tab.
+ @discussion This method informs only the specific extension of the opening of a new tab. If the intention is to inform all loaded
+ extensions consistently, you should use the respective method on the extension controller instead.
+ @seealso didCloseTab:
+ @seealso openTabs
+ */
+- (void)didOpenTab:(id <_WKWebExtensionTab>)newTab;
+
+/*!
+ @abstract Should be called by the app when a tab is closed to fire appropriate events with only this extension.
+ @param closedTab The tab that was closed.
+ @param windowIsClosing A boolean value indicating whether the window containing the tab is also closing.
+ @discussion This method informs only the specific extension of the closure of a tab. If the intention is to inform all loaded 
+ extensions consistently, you should use the respective method on the extension controller instead.
+ @seealso didOpenTab:
+ @seealso openTabs
+ */
+- (void)didCloseTab:(id <_WKWebExtensionTab>)closedTab windowIsClosing:(BOOL)windowIsClosing;
+
+/*!
+ @abstract Should be called by the app when a tab is selected to fire appropriate events with only this extension.
+ @param selectedTab The tab that was selected, or \c nil if no tab is selected or a tab is selected that is not visible to this extension.
+ @discussion This method informs only the specific extension that a tab has been selected. If the intention is to inform all loaded
+ extensions consistently, you should use the respective method on the extension controller instead.
+ */
+- (void)didSelectTab:(nullable id <_WKWebExtensionTab>)selectedTab;
```

Comment:
We should consider how multi-select of tabs works with this API.

Reply:
Changed this to `didSelectTabs:` and takes a `NSSet`.